src/environmentd/tests/sql.rs

```diff
@@ -3741,6 +3741,34 @@ async fn test_explain_timestamp_blocking() {
     );
 }
 
+#[mz_ore::test(tokio::test(flavor = "multi_thread", worker_threads = 1))]
+#[cfg_attr(miri, ignore)] // too slow
+async fn test_explain_timestamp_on_const_with_temporal() {
+    // Regression test for https://github.com/MaterializeInc/materialize/issues/25841
+    let server = test_util::TestHarness::default().start().await;
+    let client = server.connect().await.unwrap();
+
+    // Create a view that is a constant, but has a temporal filter.
+    client
+        .batch_execute(
+            "CREATE VIEW const_with_temporal_filters AS
+            WITH dt_series AS (
+                SELECT generate_series(
+                    date_trunc('day', date '2024-03-02'),
+                    date_trunc('day', date '2024-03-10'),
+                    '1 day'::interval
+                ) AS dt
+            )
+            SELECT dt FROM dt_series WHERE mz_now() <= dt + '7 days'::interval AND dt < mz_now();",
+        )
+        .await
+        .unwrap();
+
+    // Before fixing the bug, the following EXPLAIN TIMESTAMP was returning u64:MAX.
+    let timestamp = get_explain_timestamp("const_with_temporal_filters", &client).await;
+    assert!(timestamp < 32503716000000); // year 3000
```

Comment:
Why the year 3000? Wouldn't `assert_ne!(timestamp, Timestamp::MAX);` be sufficient?

Reply:
I wasn't sure whether we'll always keep choosing simply `Timestamp::MAX` for `NoTimestamp` or this can somehow come out to `Timestamp::MAX-1` or something like that.